mlflow/sklearn/__init__.py

```diff
@@ -1612,6 +1651,31 @@ def patched_predict(original, self, *args, **kwargs):
             _AUTOLOGGING_METRICS_MANAGER.register_prediction_result(
                 run_id, eval_dataset_name, predict_result
             )
+            if log_datasets:
```

Comment:
@prithvikannan Can we log the datasets before model training? This way, users can quickly tell what datasets are being used for training even if the training takes hours, which can happen in the case of big tensorflow models. This should just involve moving the code up for each autologging integration :)